src/transactions.rs

```diff
@@ -0,0 +1,456 @@
+            txn_by_hash,
+            expiring_txns,
+            uids_awaiting_memos,
+        })
+    }
+
+    /// Iterate through the assets.
+    pub fn iter(&self) -> impl Iterator<Item = Transaction<L>> + '_ {
+        self.store.iter().cloned()
+    }
+
+    /// Get the asset by the code from the store.
+    pub fn get(&self, uid: &TransactionUID<L>) -> Result<Transaction<L>, KeystoreError<L>> {
+        Ok(self.store.load(uid)?)
+    }
+
+    /// Get a mutable asset editor by the code from the store.
+    pub fn get_mut(
+        &mut self,
+        uid: &TransactionUID<L>,
+    ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
+        let txn = self.get(uid)?;
+        Ok(TransactionEditor::new(&mut self.store, txn))
+    }
+
+    /// Get a Transaction with the id of a memo, will return the transaction which is awaiting the memo
+    pub fn with_memo_id(&self, id: u64) -> Result<Transaction<L>, KeystoreError<L>> {
+        let uid = self
+            .uids_awaiting_memos
+            .get(&id)
+            .ok_or(KeyValueStoreError::KeyNotFound)?;
+        Ok(self.get(uid).unwrap())
+    }
+
+    /// Same as with_memo_id but return a TransactionEditor
+    pub fn with_memo_id_mut(
+        &mut self,
+        id: u64,
+    ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
+        let txn = self.with_memo_id(id)?;
+        Ok(TransactionEditor::new(&mut self.store, txn))
+    }
+
+    /// Get a Transaction for a given TransactionHash
+    pub fn with_hash(&self, hash: &TransactionHash<L>) -> Result<Transaction<L>, KeystoreError<L>> {
+        let uid = self
+            .txn_by_hash
+            .get(hash)
+            .ok_or(KeyValueStoreError::KeyNotFound)?;
+        Ok(self.get(uid).unwrap())
+    }
+
+    /// Get a TransactionEditor for a given TransactionHash
+    pub fn with_hash_mut(
+        &mut self,
+        hash: &TransactionHash<L>,
+    ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
+        let txn = self.with_hash(hash)?;
+        Ok(TransactionEditor::new(&mut self.store, txn))
+    }
+
+    /// Get all the transactions timing out at the provided time.  There is no mutable way to get this
+    /// Collection because we should only be edditing one thing at a time
+    pub fn with_timeout(&self, timeout: u64) -> Result<Vec<Transaction<L>>, KeystoreError<L>> {
+        let uids = self
+            .expiring_txns
+            .get(&timeout)
+            .ok_or(KeyValueStoreError::KeyNotFound)?;
+
+        Ok(uids.iter().map(|uid| self.get(uid).unwrap()).collect())
+    }
+
+    /// Commit the store version.
+    pub fn commit(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.commit_version()?)
+    }
+
+    /// Revert the store version.
+    pub fn revert(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.revert_version()?)
+    }
+
+    /// Add a transaction has to the index and update the stored transaction with this has
+    pub fn insert_hash(
```

Comment:
Is there a need for this to be a public function? For me, this seems like it should just be something that automatically happens when calling `save` after a create or update operation.